compiler/rustc_ast/src/util/classify.rs

```diff
@@ -113,6 +113,10 @@ pub fn expr_trailing_brace(mut expr: &ast::Expr) -> Option<&ast::Expr> {
             | While(..)
             | ConstBlock(_) => break Some(expr),
 
+            Cast(_, ty) => {
+                break type_trailing_brace(ty).then_some(expr);
+            }
```

Comment:
Only `ast::TyKind::MacCall` will cause this to end with `}`(`ast::TyKind::AnonStruct` and `ast::TyKind::AnonStruct` will only be used in struct fields). So it will be more accurate to report the error using `errors::WrapInParentheses::MacroArgs` in [`check_let_else_init_trailing_brace`](https://github.com/rust-lang/rust/blob/master/compiler/rustc_parse/src/parser/stmt.rs#L411).

Reply:
Good call. I have implemented this improvement.